addons/langfuse/templates/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "langfuse.fullname" . }}
+  labels:
+    {{- include "langfuse.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "langfuse.selectorLabels" . | nindent 6 }}
```

Comment:
`matchLabels` are immutable, which means if you ever change the input to langfuse.selectorLabels, then you'll have to delete and recreate the depoyment.

I'd consider breaking out podLabels vs matchLabels